js/reference.js

```diff
@@ -1555,6 +1560,83 @@ var ERMrest = (function(module) {
             return newReference;
         },
 
+        /**
+         * NOTE: If an alias for an aggregate is a duplicate of another provided alias, a new one will be generated
+         *
+         * @param {ERMrest.ColumnAggregateFn[]} aggregateList - list of aggregate functions to apply to GET uri
+         * @return {Promise} - Promise contains an object in the form of {alias: value, alias: value}
+         */
+        getAggregates: function(aggregateList) {
+            var defer = module._q.defer();
+            var url;
+
+            var URL_LENGTH_LIMIT = 2048;
+
+            var urlSet = [];
+            var baseUri = this.location.service + "/catalog/" + this.location.catalog + "/aggregate/" + this.location.ermrestCompactPath + "/";
+            // for matching the aliases after the data returns
+            var aggAliases = [];
+            for (var i = 0; i < aggregateList.length; i++) {
+                var agg = aggregateList[i];
+                var alias = (agg.alias != null ? agg.alias : i);
+
+                // this alias is already used, should only be the case when the user defined an alias to be used twice
+                if (aggAliases.indexOf(alias) > -1) {
```

Comment:
Instead of just an `if`, it should be recursive. If `aggregateList = [ {'alias': 1, ..}, {<no-alias> ...}]` it's still going to use the same alias for both of them.